libr/debug/debug.c

```diff
@@ -848,15 +1015,25 @@ R_API int r_debug_continue_syscalls(RDebug *dbg, int *sc, int n_sc) {
 		return -1;
 	}
 	for (;;) {
+		RDebugReasonType reason;
+
 		if (r_cons_singleton()->breaked)
 			break;
 #if __linux__
 		// step is needed to avoid dupped contsc results
+		/* XXX(jjd): actually one stop is before the syscall, the other is
+		 * after.  this allows you to inspect the arguments before and the
+		 * return value after... */
 		r_debug_step (dbg, 1);
 #endif
 		dbg->h->contsc (dbg, dbg->pid, 0); // TODO handle return value
 		// wait until continuation
-		r_debug_wait (dbg);
+		reason = r_debug_wait (dbg);
+		if (reason == R_DEBUG_REASON_DEAD || r_debug_is_dead (dbg))
```

Comment:
One is a flag and one tries ```kill(pid, 0)```. I am trying to avoid doing an extra syscall here for no reason.